src/AspNet.Security.OAuth.BattleNet/BattleNetAuthenticationOptions.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.AspNet.Authentication.OAuth;
+using Microsoft.AspNet.Http;
+using System;
+
+namespace AspNet.Security.OAuth.BattleNet {
+    /// <summary>
+    /// Defines a set of options used by <see cref="BattleNetAuthenticationHandler"/>.
+    /// </summary>
+    public class BattleNetAuthenticationOptions : OAuthAuthenticationOptions {
+        public BattleNetAuthenticationOptions() {
+            AuthenticationScheme = BattleNetAuthenticationDefaults.AuthenticationScheme;
+            Caption = BattleNetAuthenticationDefaults.Caption;
+            ClaimsIssuer = BattleNetAuthenticationDefaults.Issuer;
+
+            CallbackPath = new PathString(BattleNetAuthenticationDefaults.CallbackPath);
+
+            this.SetServerRegion(ServerRegion.US); //Default to US
+
+            UserInformationEndpoint = BattleNetAuthenticationDefaults.UserInformationEndpoint;
+
+            SaveTokensAsClaims = false;
+        }
+
+        public void SetServerRegion(ServerRegion serverRegion) {
```

Comment:
Would it not be better to have a property called `ServerRegion` and then automatically set the correct endpoints in the setter of that property, rather than having the user call a method?

The normal pattern for setting options is always to just set a property, but now we are deviating from this by having the user calling a method.